tmt/steps/execute/__init__.py

```diff
@@ -436,6 +436,9 @@ def check_custom_results(self, test: "tmt.Test", guest: Guest) -> List["tmt.Resu
             # can be useful, for grouping results that belong to the same tests.
             partial_result.serialnumber = test.serialnumber
 
+            # Enforce the correct guest info
+            partial_result.guest = ResultGuestData(name=guest.name, role=guest.role)
```

Comment:
Shouldn't we store data from the custom results? 
I see the test results.yaml has `client-2` but the final results.yaml available in execute has `default-0`

Reply:
That's intentional, and I suppose both ways have their pros and cons. As I see it, `client-2` is clearly incorrect, as the test was executed on `default-0`, which was a local host. In fact, there's no `client-2` in this run at all, if I see `client-2` here and peek into `guests.yaml`, I would not find it there. Therefore my goal here was to assure consumers of `results.yaml` (and `guests.yaml` and `tests.yaml` as well) would get a consistent eventual picture, by enforcing `results.yaml` to contain information tmt "owns" and knows for sure is correct: `serialnumber`, `guest` and, in https://github.com/teemtee/tmt/pull/1911, also time-related fields like `duration`. I don't try to touch logs or the outcome, just the "metadata" bits that do not represent the test result.

The other option would be to treat the input as sacred and let it pass, but for me, that opens the door for tests to inject wrong and misleading metadata. For example, any piece of tmt code inspecting collected results in the `report` step would have to be able to deal with the situation the guest, referenced from a particular result, is unknown and does not exist, or that results, sorted by their serial numbers, do not match the tests sorted by their serial numbers, because custom results contained a typo in this field.